src/components/auction/AuctionDetails/index.tsx

```diff
@@ -210,20 +212,38 @@ const AuctionDetails = (props: Props) => {
         derivedAuctionInfo?.biddingToken,
         derivedAuctionInfo?.auctioningToken,
       )
-    const clearingPriceNumber = orderToPrice(clearingPriceInfoAsSellOrder)?.toSignificant(4)
+    const clearingPriceNumber =
+      showPriceInverted && orderToPrice(clearingPriceInfoAsSellOrder)
+        ? orderToPrice(clearingPriceInfoAsSellOrder).invert().toSignificant(5)
+        : orderToPrice(clearingPriceInfoAsSellOrder)?.toSignificant(5)
```

Comment:
`orderToPrice` seems like is having two responsibilities, checking if that order has a "calculable" price and calculating it. So I think the first call to it is unnecessary because then is used with `?.` operator. Now is possible to have `showPriceInverted` as false and the first call to `orderToprice` not called. Another option is to extract everything as `priceFromOrder` or something similar.